packages/pipelines/src/commands/pipelines/promote.ts

```diff
@@ -121,19 +127,47 @@ async function streamReleaseCommand(heroku: APIClient, targets: Array<Heroku.App
   }
 
   cli.log('Running release command...')
-  const fetch = (retry = 100) => new Promise((resolve, reject) => {
-    const stream = got.stream(release.output_stream_url!)
-    stream.on('error', async err => {
-      await wait(100)
-      if (retry && err.code === '404') {
-        fetch(retry - 1).then(resolve).catch(reject)
-      } else reject(err)
+
+  async function streamReleaseOutput(releaseStreamUrl: string) {
+    const finished = util.promisify(Stream.finished)
+    const fetchResponse = await fetch(releaseStreamUrl)
+
+    if (fetchResponse.status >= 400) {
+      throw new Error('stream release output not available')
+    }
+
+    fetchResponse.body.pipe(process.stdout)
+
+    const promise = new Promise((resolve, reject) => {
+      return finished(fetchResponse.body)
+      .then(() => resolve(''))
+      .catch(error => reject(error))
     })
-    stream.on('end', resolve)
-    const piped = stream.pipe(process.stdout)
-    piped.on('error', reject)
+
+    return promise
```

Comment:
You don't need to wrap with a promise here to return the string.

```javascript
await finished(fetchResponse.body)
```

We don't use the return value of the empty string anywhere so we can leave it out.